blockchain/utxobackend.go

```diff
@@ -171,72 +305,160 @@ func removeRegressionDB(net wire.CurrencyNet, dbPath string) error {
 	return nil
 }
 
+// fileExists reports whether the named file or directory exists.
+func fileExists(name string) bool {
+	if _, err := os.Stat(name); err != nil {
+		if os.IsNotExist(err) {
+			return false
+		}
+	}
+	return true
```

Comment:
It's probably worth revisiting this now with newer versions of Go.

Once upon a time when things that dealt with files like this was written (almost a decade ago at this point), there was not a 100% reliable way to detect the exact error from `os.Stat` as in-depth testing showed that it varied per OS and was rather inconsistent (e.g. UNC paths, NFS mounts, "missing" due to permissions errors, plan9 strangeness, etc), so the current approach was adopted as a happy medium that works reasonably well everywhere and we've been using it ever since.

That said, it has worked quite well for almost a decade now, so while I'm definitely not opposed to revisiting this with all the associated testing across all OSes, even the ones that aren't officially supported, I would suggest such a change, if desired, really should be in a separate PR.

Reply:
Yeah, I discussed this a bit with @davecgh, and we opted to keep the interactions with `os.Stat` and `os.MkdirAll` the same as they are in the path for the overall database since that code has been around for a while and proven out across many platforms.  There may very well be a more robust solution, but it would require a lot more digging and testing.

Reply:
I figured it was out of scope here, but wanted to bring it up. :+1: